rpmlint/checks/TmpfilesParser.py

```diff
@@ -0,0 +1,166 @@
+    entry."""
+
+    def __init__(self, source, line, comments):
+        """Parses the given tmpfiles configuration line and stores the
+        resulting information in the object.
+
+        `source` is the configuration file name, `line` the line number of the
+        entry to be parsed and `comments` is a list of context lines that
+        likely belong to this entry."""
+        self.source = source
+        self.line = line
+        self.comments = comments
+        self.valid = False
+        self.warnings = []
+        self._parse()
+
+    def _add_warning(self, msg):
+        self.warnings.append(f'{self.source}:{self.line}: {msg}')
+
+    def get_warnings(self):
+        """Returns a list of parse errors/warnings that have been encountered
+        during parsing the tmpfiles entry."""
+        return self.warnings
+
+    def _parse(self):
+        # each field may be quoted, so use shell like parsing to cope with that
+        fields = shlex.split(self.line)
+        if len(fields) > 7:
+            self._add_warning('Too many fields encountered')
+            return
+        elif len(fields) < 2:
+            self._add_warning('Too few fields encountered')
+            return
+
+        self.type = fields[0]
+        self.target = fields[1]
+
+        findex = 2
+
+        def next_field():
+            nonlocal findex
+            if findex >= len(fields):
+                return None
+
+            ret = fields[findex]
+            findex += 1
+            return ret if ret != '-' else None
+
+        self.mode = next_field()
+        self.owner = next_field()
+        self.group = next_field()
+        self.age = next_field()
+        self.arg = next_field()
+
+        if (self.mode or self.owner or self.group) and not self._supports_perms():
+            self._add_warning("Permissions specified for entry type that doesn't support perms")
+
+        self._normalize()
+        self.valid = True
+
+    def is_valid(self):
+        """Returns whether a valid entry could be parsed."""
+        return self.valid
+
+    def _normalize(self):
+        """Normalizes any data of the current entry for simplification."""
+        if self.type[0] == 'F':
+            # see _supports_perms()
+            self.type = 'f+' + self.type[1:]
+
+    def _supports_perms(self):
+        """Returns whether the current tmpfiles entry supports a file
+        permissions field."""
+        # NOTE: for 'C' the table in the man page does not list permissions
+        # support but actually they *are* supported.
+        # NOTE: F is a deprecated form of 'f+'
+        return self.type[0] in ('f', 'F', 'd', 'D', 'e', 'v', 'q', 'Q', 'p', 'p+', 'c', 'C', 'b', 'z', 'Z')
+
+    def _get_def_mode_label(self):
+        return '-' if self._supports_perms() else 'N/A'
+
+    def get_type(self):
+        return self.type
+
+    def get_config_file(self):
+        return self.source
+
+    def get_target_path(self):
+        return self.target
```

Comment:
Do we really need this getters?

Reply:
Not at the moment but I'd like the types's interface to be complete.